ast/src/parsed/asm.rs

```diff
@@ -109,6 +110,13 @@ impl SymbolPath {
         self.parts.extend(other.into().parts);
         self
     }
+
+    /// Formats the path and uses `.` as separator if
+    /// there are at most two components.
+    pub fn to_dotted_string(&self) -> String {
+        let separator = if self.parts.len() <= 2 { "." } else { "::" };
```

Comment:
This feels a bit confusing. Maybe we should keep `::` for namespacing and static access, and `.` for instance access? (if any of that makes sense)

Reply:
I think currently we don't have any instances in pil. When we add structs, then we should use `.` for field access and `::` for namespaces, yes.